src/Microsoft.ML.ImageAnalytics/ImageLoaderTransform.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.IO;
+using System.Text;
+using Microsoft.ML.Runtime.ImageAnalytics;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(ImageLoaderTransform.Summary, typeof(ImageLoaderTransform), typeof(ImageLoaderTransform.Arguments), typeof(SignatureDataTransform),
+    ImageLoaderTransform.UserName, "ImageLoaderTransform", "ImageLoader")]
+
+[assembly: LoadableClass(ImageLoaderTransform.Summary, typeof(ImageLoaderTransform), null, typeof(SignatureLoadDataTransform),
+   ImageLoaderTransform.UserName, ImageLoaderTransform.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.ImageAnalytics
+{
+    // REVIEW: Rewrite as LambdaTransform to simplify.
+    /// <summary>
+    /// Transform which takes one or many columns of type <see cref="DvText"/> and loads them as <see cref="ImageType"/>
+    /// </summary>
+    public sealed class ImageLoaderTransform : OneToOneTransformBase
+    {
+        public sealed class Column : OneToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                Contracts.AssertNonEmpty(str);
+
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public sealed class Arguments : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)",
+                ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Folder where to search for images", ShortName = "folder")]
+            public string ImageFolder;
+        }
+
+        internal const string Summary = "Load images from a file.";
```

Comment:
>a file [](start = 58, length = 6)

You forgot to pluralize this.

Reply:
is "a files" ok, or it should be something else?

---
In reply to: [203182343](https://github.com/dotnet/machinelearning/pull/528#discussion_r203182343) [](ancestors = 203182343)

Reply:
Correct would be "load images from files."

---
In reply to: [205597457](https://github.com/dotnet/machinelearning/pull/528#discussion_r205597457) [](ancestors = 205597457,203182343)